core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -177,8 +178,8 @@ private TableMetadata getTableMetadata(Scan scan) throws ExecutionException {
     }
   }
 
-  public Optional<List<Key>> get(Scan scan) {
-    if (isWriteSetOverlappedWith(scan)) {
+  public Optional<List<Key>> get(ScanInterface scan) {
+    if (scan instanceof Scan && isWriteSetOverlappedWith((Scan) scan)) {
```

Comment:
This check (or an equivalent) won't be needed for RelationalScan?

Reply:
Good question. A check for preventing scan-after-write is necessary for RelationalScan too. But it's done at a bit different timing, and the PR will be proposed later.